### DISCUSSION TITLE
Using --follow-fork with multiprocessing forkserver

### DISCUSSION THREAD

**Discussion by dkbarn** [bloomberg/memray#572] [Q&A]
I am trying to profile an application that uses a ProcessPoolExecutor set to "forkserver" mode. I am making sure to pass --follow-fork to memray, however, I only get a single .bin file, and no information whatsoever for the processes in the pool. Here's a simplified example of the code:

```python
import multiprocessing
from concurrent.futures import ProcessPoolExecutor, as_completed

mp_context = multiprocessing.get_context("forkserver")
with ProcessPoolExecutor(mp_context=mp_context) as executor:
    futures = [executor.submit(my_fn) for _ in range(100)]
    for future in as_completed(futures):
        result = future.result()
```

Should this be expected to work?

**Comment by pablogsal** [comment]
> Should this be expected to work?

No, unfortunately we can only track direct forks of the application itself. Check https://bloomberg.github.io/memray/run.html#tracking-across-forks for more information.
